### PR TITLE
Recovery mode

### PR DESCRIPTION
**Pull request by mrsmkl**
Started working on recovery mode. Work in progress, but nothing should be broken.

### CODE REVIEW

**Comment by mrsmkl** [comment]
The idea is to check events from previous blocks to find out what the clients should do.

**Comment by hswick** [comment]
It would be great to have some documentation in the README on how to demo the recovery option.

I tried running the client with the command line, starting a task, submitting solution, then turning it off and on and again and trying to resume the task, but didnt seem to work.

**Comment by hswick** [comment]
Which phases have you tested it on? I'm able to get the recovery to start if I set the write number of blocks.

However, testing it during the challenge period, the solver keeps solving the task again and then tries to submit the solution again, which results in a revert. Looks like it needs some kind of logic to check if it submitted the solution already.

File: cli/index.js [hunk 1]
```diff
@@ -28,19 +28,22 @@ vorpal
   })
 
 vorpal
-  .command('start <command>', 'start a task giver.')
+  .command('start <command>', 'start a task giver, solver or verifier [task,solve,verify].')
   .option('-a, --account <num>', 'index of web3 account to use.')
+  .option('-r, --recover <num>', 'recovery mode: check the blocks.')
```
[the remark this answers is not in the document]
Comment (mrsmkl) [reply cli/index.js]: It should read: check <num> blocks

**Comment by mrsmkl** [comment]
The logic is that it handles the last event associated with the task. So for example a listener for `SolutionsCommitted` event would have to be added to the solver. But if it tried to submit at verification game, then there is an error because there was `EndRevealPeriod` event. But check that this wasn't another task posted within those blocks.

**Comment by hswick** [comment]
This is great. I tried it out on before and after challenge period and got it to work.

**Pull request merged**